### PR TITLE
feat: support maximum quantity displayed and expand more logic for @ menu

### PR DESCRIPTION
**Pull request by akumatus**
Fix issue [BS-790](https://linear.app/affine-design/issue/BS-790).

<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/sJGviKxfE3Ap685cl5bj/b9186764-ef44-4860-87b0-8d4da775d681.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/sJGviKxfE3Ap685cl5bj/b9186764-ef44-4860-87b0-8d4da775d681.mov">录屏2024-07-10 19.01.15.mov</video>

### What Changed?
- Add `EdgelessIcon` for @ menu list
- Support maximum quantity displayed and expand more logic for @ menu
- Add e2e test for expand more docs

### CODE REVIEW

**Comment by akumatus** [comment]
* **#7540** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7540?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akumatus and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/blocks/src/root-block/widgets/linked-doc/config.ts [hunk 1]
```diff
@@ -79,16 +89,20 @@ export const getMenus: (ctx: {
   const filteredDocList = docMetas
     .filter(({ id }) => id !== doc.id)
     .filter(({ title }) => isFuzzyMatch(title, query));
+  const MAX_DOCS = 6;
```
Comment (L-Sun) [inline packages/blocks/src/root-block/widgets/linked-doc/config.ts:92]: I think it would be better to make `maxDisplay` configurable in `LinkedDocOptions`

**Review by L-Sun — APPROVED** [review]

File: packages/blocks/src/root-block/widgets/linked-doc/config.ts (hunk 1 above)
Comment (akumatus) [reply packages/blocks/src/root-block/widgets/linked-doc/config.ts:92]: Thanks! Maybe this will be configurable inside block specs in the feature.

**Review by Svaney-ssman — APPROVED** [review]
GooGood 👍

**Pull request merged**